File: src/main/java/cn/guajichi/idlepool/util/TimeFormats.java
package cn.guajichi.idlepool.util;

public final class TimeFormats {
    private TimeFormats() {
    }

    public static String clock(long totalSeconds) {
        long safe = Math.max(0, totalSeconds);
        long hours = safe / 3_600;
        long minutes = (safe % 3_600) / 60;
        long seconds = safe % 60;
        return "%02d:%02d:%02d".formatted(hours, minutes, seconds);
    }
}

File: src/main/kotlin/top/cnuo/idlepool/util/TimeFormats.kt
package top.cnuo.idlepool.util

object TimeFormats {
    @JvmStatic
    fun clock(totalSeconds: Long): String {
        val safe = totalSeconds.coerceAtLeast(0)
        return "%02d:%02d:%02d".format(safe / 3600, (safe % 3600) / 60, safe % 60)
    }
}
